NitroxPatcher/Patches/Persistent/EntityCell_AwakeAsync_Patch.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using Harmony;
+using NitroxModel.Helper;
+using NitroxClient.MonoBehaviours;
+
+namespace NitroxPatcher.Patches.Persistent
+{
+    public class EntityCell_AwakeAsync_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(EntityCell);
+
+        private static readonly FieldInfo current = getLoadAsyncEnumerableMethod().GetField("$current", BindingFlags.Instance | BindingFlags.NonPublic);
+
+        public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> codeInstructions = instructions.ToList();
+            yield return new CodeInstruction(OpCodes.Ldarg_0);
+            yield return new CodeInstruction(OpCodes.Ldnull);
+            yield return new CodeInstruction(OpCodes.Stfld, current);
+
+            int i = codeInstructions.IndexOf(codeInstructions.Last()) - 3;
+            Label label = generator.DefineLabel();
+            codeInstructions[i].labels.Add(label);
+            yield return new CodeInstruction(OpCodes.Br, label);
+
+            foreach (CodeInstruction instruction in codeInstructions)
+            {
+                yield return instruction;
+            }
+        }
+
+        public override void Patch(HarmonyInstance harmony)
+        {
+            PatchTranspiler(harmony, getMethod());
+        }
+
+        private static Type getLoadAsyncEnumerableMethod()
+        {
+            Type[] nestedTypes = TARGET_CLASS.GetNestedTypes(BindingFlags.NonPublic | BindingFlags.Static);
+            Type targetEnumeratorClass = null;
+
+            foreach (Type type in nestedTypes)
+            {
+                if (type.FullName.Contains("AwakeAsync"))
+                {
+                    targetEnumeratorClass = type;
```

Comment:
```CS
Validate.NotNull(type)
return type;
```

Reply:
I believe it was set up in this manner to throw an error when the type could not be found